### PR TITLE
Add pip support to build system | `pip install pymoose` will work on all platforms

### PR DESCRIPTION
**Pull request by dilawar**
## TL/DR summary 

Current build instructions, 
```
$ mkdir _build
$ cmake ..
$ make install
```
After this PR merge, these will reduce to

```
$ pip install pymoose     # to install stable version
$ pip install git+https://github.com/bhallalab/moose-core # for bleeding edge 
```

# Details 

Currently `moose-core` is built by `cmake`. The instructions are tedious for normal python users.

Python users are more accustomed to `python setup.py install` when building from the source code. This PR adds a build system based on `python-setuptool`. Note that this does not remove `cmake` based build system; in-fact `setup.py` uses `cmake` to build the `_moose.so`. Old instructions are still valid. `cmake` is preferred for developers and testing with various macro which `setup.py` is for normal user with default settings.

Adding `setup.py` based build system gives us another advantage. If `pip install pymoose` does not find appropriate wheels (binary build); it will pull the source code and build locally. For example, the following one-liner will work (if this PR passes) on almost all cases now:

    $ pip install git+https://github.com/bhallalab/moose-core --user

### CODE REVIEW

**Comment by dilawar** [comment]
@upibhalla @hrani Please review. This is ready to be merged.

After this is merged, I'll update the deploy scripts which automatically upload the built binaries and other artifacts to appropriate places.
